src/OrchardCore/OrchardCore.Abstractions/BackgroundJobs/HttpBackgroundJob.cs

```diff
@@ -76,6 +78,10 @@ public static Task ExecuteAfterEndOfRequestAsync(string jobName, Func<ShellScope
                 var logger = scope.ServiceProvider.GetRequiredService<ILogger<ShellScope>>();
                 try
                 {
+                    var nextHttp = scope.ServiceProvider.GetRequiredService<IHttpContextAccessor>();
+                    //Restore user credentials
+                    nextHttp.HttpContext.User = userPrincipal;
```

Comment:
Why not have that just after ln 63?

Reply:
Because after that we'll be fetching a new http context, and I don't think we'll be able to get the right login info at that point.

Well, I don't remember if this was useful in the previous code, but for now it doesn't seem to be necessary to use the login user context in the existing code reference.

This PR is submitted in order to start the workflow asynchronously in the script, because this will create a new http context, which will make it impossible to retrieve user information

Reply:
The `HttpContext` object here should be the same across all scopes, otherwise, ln 63 wouldn't make sense. Please check.

Can you please explain what you mean by "for now it doesn't seem to be necessary to use the login user context in the existing code reference"? Is this change not useful anymore?

Reply:
No, not really, as you can see in the code below.

https://github.com/OrchardCMS/OrchardCore/blob/eaf58d000017f65ed650804ac35867371bfb558e/src/OrchardCore/OrchardCore.Abstractions/BackgroundJobs/HttpBackgroundJob.cs#L38-L50

I mean, the original reason for changing this API is that I have a custom API that uses this function to trigger the workflow asynchronously rather than having to wait for the workflow to complete before returning the request result. But the newly obtained  `HttpContext.User` is null

Reply:
There is really no such use case in OC at the moment

Reply:
It seems to me it is the same object:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/96942ce6-afdc-4a26-8b04-971cc1de0b9d)

Reply:
Can you also check the `HttpContext.User`  ? I remember it was empty

Reply:
I'll add a unit test to verify it

Reply:
It looks like unit test simulation is a little difficult. It requires an http request to activate.


I cloned it before it was cleared because the user's information would be wiped clean when it was retrieved
![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/e4fd0c2f-938a-4e82-b3b7-2e91c1d9ec03)

Reply:
Really all I'm asking is to not have a HCA resolution to set the user, but rather set it on line 64 via the property there. Since the `HttpContext` is the same, this should work, unless something in between nulls it out. Please, just try it out.

Reply:
I see what you mean. We've made some adjustments